pkg/authenticator/patcher.go

```diff
@@ -0,0 +1,69 @@
+package authenticator
```

Comment:
Given that this isn't part of the server, perhaps it could live in a `authenticator/patcher` package.